### PR TITLE
Add internal load balancer to GCP.

### PR DESCRIPTION
**Pull request by patrickdillon**
JIRA: [CORS-1235](https://jira.coreos.com/browse/CORS-1235)
Adds an internalTCP load balancer with instance groups containing the masters and bootstrap node(which has been removed from the external LB target pool).  Points internal DNS to use this load balancer. Due to the restriction in the internal load balancer we can only perform one health check, so no health check is being performed against the mcs. We are assuming that if the api is alive and ready, so is mcs.

### CODE REVIEW

File: data/data/gcp/network/firewall.tf [hunk 1]
```diff
@@ -22,12 +22,8 @@ resource "google_compute_firewall" "mcs" {
     ports    = ["22623"]
   }
 
-  source_ranges = var.preexisting_network ? ["0.0.0.0/0"] : [
-    var.network_cidr,
-    google_compute_address.master_nat_ip[0].address,
-    google_compute_address.worker_nat_ip[0].address
-  ]
-  target_tags = ["${var.cluster_id}-master"]
+  source_ranges = var.preexisting_network ? ["0.0.0.0/0"] : [var.network_cidr]
```
Comment (abhinavdahiya) [inline data/data/gcp/network/firewall.tf]: i think now, it should always be the network_cidr

File: pkg/destroy/gcp/compute.go [hunk 2]
```diff
@@ -111,6 +111,29 @@ func (o *ClusterUninstaller) deleteInstanceGroup(ig nameAndZone) error {
 	return nil
 }
 
+// destroyInstanceGroups searches for instance groups across all zones that have a name that starts with
```
Comment (abhinavdahiya) [inline pkg/destroy/gcp/compute.go]: I think we also need to make sure health-checks (ILB), http-health-checks (NLB) are destroyed.

File: data/data/gcp/network/firewall.tf (hunk 1 above)
Comment (abhinavdahiya) [reply data/data/gcp/network/firewall.tf]: also we need to make sure the source range for health checks for ILB are added here https://github.com/openshift/installer/blob/39c63115aaf52181323d8b76807c3469465969cc/data/data/gcp/network/firewall.tf#L39

**Comment by abhinavdahiya** [comment]
https://github.com/openshift/installer/blob/39c63115aaf52181323d8b76807c3469465969cc/pkg/asset/machines/gcp/machines.go#L112

The master machines have reference to a dropped resource, which is making it fail when updating status of the machine.

> https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/2448/pull-ci-openshift-installer-master-e2e-gcp/85/artifacts/e2e-gcp/must-gather/registry-svc-ci-openshift-org-ci-op-v42w6l40-stable-sha256-8f5a70fa17fe774580a1431f3e987d8bdad27f3cfe322b5a590548926095a2ab/namespaces/openshift-machine-api/pods/machine-api-controllers-7c7f6c6568-d55gr/machine-controller/machine-controller/logs/current.log
`2019-10-02T23:35:41.198445043Z E1002 23:35:41.195866       1 controller.go:246] Error updating machine "openshift-machine-api/ci-op--tv7q9-m-2": unable to get targetpool: googleapi: Error 404: The resource 'projects/openshift-gce-devel-ci/regions/us-east1/targetPools/ci-op--tv7q9-ign' was not found, notFound
`

File: data/data/gcp/network/firewall.tf (hunk 1 above)
Comment (patrickdillon) [reply data/data/gcp/network/firewall.tf]: >also we need to make sure the source range for health checks for ILB are added here

I had not considered this, but it looks like we are covered. Those ranges cover all load balancers except internal https: https://cloud.google.com/load-balancing/docs/health-checks#firewall_rules

File: pkg/destroy/gcp/compute.go (hunk 2 above)
Comment (patrickdillon) [reply pkg/destroy/gcp/compute.go]: > I think we also need to make sure health-checks (ILB), http-health-checks (NLB) are destroyed.

I think they are covered here: https://github.com/openshift/installer/pull/2448/files/39c63115aaf52181323d8b76807c3469465969cc#diff-4781d154886c4da10a007bb219f27d8aR140-L140

I have noticed any leaked resources in my personal install/deletes.

**Pull request merged**